rpy2/robjects/functions.py

```diff
@@ -239,3 +243,124 @@ def __doc__(self):
+_SCALAR_COMPAT_RTYPES = set(
+    getattr(rinterface.RTYPES, name).value
+    for name in ('STRSXP', 'INTSXP', 'REALSXP', 'LGLSXP', 'CPLXSXP')
+)
+
+
+def _map_default_value(value: rinterface.Sexp):
+    """
+    Map default in the R signature.
+
+    Because of R's lazy evaluation some default might be unevaluated expressions.
+
+    Args:
+      value:
+    """
+    if value.__sexp__.typeof in _SCALAR_COMPAT_RTYPES:
+        res = value[0]
+    else:
+        res = value
+    return res
+
+
+def map_signature(
+        r_func: SignatureTranslatedFunction,
+        method_of: bool = False,
+        map_default: typing.Optional[
+            typing.Callable[[rinterface.Sexp], typing.Any]
+        ] = _map_default_value
+) -> inspect.Signature:
+    """
+    Map the signature of an function to the signature of a Python function.
+
+    Args:
+        r_func (SignatureTranslatedFunction): an R function
+        method_of (bool): Whether the function should be treated as a method
+            (adds a `self` param to the signature if so).
+        map_default (function): Function to map default values in the Python
+            signature. No mapping to default values is done if None.
+    Returns:
+        An inspect.Signature.
+    """
+    params = []
+    if method_of:
+        params.append(inspect.Parameter('self',
+                                        inspect.Parameter.POSITIONAL_ONLY))
+    r_params = r_func.formals()
+    rev_prm_transl = {v: k for k, v in r_func._prm_translate.items()}
+    if r_params.names is not rinterface.NULL:
+        for name, default_orig in zip(r_params.names, r_params):
+            if name == '...':
+                warnings.warn('The R ellispsis is not yet supported.')
+            transl_name = rev_prm_transl.get(name)
+            default_orig = default_orig[0]
+            if map_default and not rinterface.MissingArg.rsame(default_orig):
+                default_mapped = map_default(default_orig)
+            else:
+                default_mapped = inspect.Parameter.empty
+            prm = inspect.Parameter(
+                transl_name if transl_name else name,
+                inspect.Parameter.POSITIONAL_OR_KEYWORD,
+                default=default_mapped
+            )
+            params.append(prm)
+    return inspect.Signature(params)
+
+
+def wrap_r_function(
+        r_func: SignatureTranslatedFunction, name: str, *,
+        method_of: bool = False, full_repr: bool = False,
+        map_default: typing.Callable[[rinterface.Sexp], typing.Any] = _map_default_value
+) -> typing.Callable:
+    """
+    Wrap an rpy2 function handle with a Python function with a matching signature.
+
+    Args:
+        r_func (rpy2.robjects.functions.SignatureTranslatedFunction): The
+            function to be wrapped.
+        name (str): The name of the function.
+        method_of (bool): Whether the function should be treated as a method
```

Comment:
In the original version `method_of` was an R6 R object (just in case we wanted to extract information from the host instance at some point). In this version, if you only accept bool values, perhaps it should be renamed `is_method`?

Reply:
Oh, I missed that bit. Thanks for catching that. 

I am trying to separate the many good things you had in your PR to:
- make it able to accommodate some of the weird or idiosyncratic R code that is out there (you might be blessed with good coding practices where you are, but I'll have the issue reports ;-) )
- take R6 mapping separately (I'll send you a note about this)